### DISCUSSION TITLE
Does this library support saving the context of a conversation.

### DISCUSSION THREAD

**Discussion by MarkOdinSon** [tectalichq/public-openai-client-php#21] [Q&A] [answered]
Does this library support saving the context of a conversation with a bot (saving correspondence in a chat)?
I found out that the OpenAI chatGPT API does not save the context of the correspondence (conversation) with the AI, so the developer himself must take care of saving the following questions and answers in order to use the context as in the web interface.

There is also a problem that if the conversation is large enough, then the limit of approximately 4000 tokens (3000 words in my opinion) will be exceeded and old queries should be removed from the context.

The documentation suggests saving the context in the database or writing to a temporary file.

What do you say about this?

Because, for example, in the web interface (https://chat.openai.com/chat), the chat by default supports the context, unlike the API

**Comment by thejamescollins** [comment, marked as answer]
Hi there @MarkOdinSon,

This library does not currently have any functionality to save or access conversation context/history.

We encourage users to implement this within their own apps themselves, using the framework and/or code of their choice.

I hope that helps,

James
